src/main/java/org/tikv/common/PDClient.java

```diff
@@ -470,11 +471,12 @@ private GetMembersResponse getMembers(BackOffer backOffer, URI uri) {
   }
 
   // return whether the leader has changed to target address `leaderUrlStr`.
-  synchronized boolean trySwitchLeader(String leaderUrlStr) {
-    if (pdClientWrapper != null) {
-      if (leaderUrlStr.equals(pdClientWrapper.getLeaderInfo())) {
+  @VisibleForTesting
+  public boolean trySwitchLeader(String leaderUrlStr) {
```

Comment:
This method is not Thread-safe, since the value of pdClientWrapper can be changed during statements